File: src/Collectors/Collector/MessageQueue.cs
namespace Collector
{
    using System;
    using System.Text;
    using RabbitMQ.Client;
    using static Configuration;

    public class MessageQueue
    {
        private IConnection _connection;

        private string _routingKey;

        private string _exchange;

        public void Initialize()
        {
            var configuration = string.Empty;

            _connection = new ConnectionFactory
            { 
                HostName = Host,
                UserName = User, 
                Password = Password, 
                Port = Port,
                VirtualHost = VirtualHost,
            }.CreateConnection();
        
            Log.Write("MessageQueue", "Initialized message queue.");
        }

        public void Finalize()
        {
            _connection.Dispose();

            _connection = null;

            Log.Write("MessageQueue", "Finalized message queue.");
        }

        public void Send(string json)
        {
            if (_connection == null)
                throw new InvalidOperationException($"Connection '{_connection}' not initialized.");

            using (var channel = _connection.CreateModel())
            {
                var bytes = Encoding.UTF8.GetBytes(json);

                channel.BasicPublish(_exchange, _routingKey, true, null, bytes);
                
                Log.Write("MessageQueue", $"Sent message: '{json}'.");
            }
        }
    }
}

File: src/Processors/Twitter/Integrations/MessageQueue.py
import pika
from Configurations import Configuration
from logzero import logger

class MessageQueue:

    def __init__(self):
        self.configuration = Configuration.MessageQueue()        

    def Initialize(self):
        credentials = pika.PlainCredentials(self.configuration['User'], self.configuration['Password'])

        self.connection = pika.BlockingConnection(pika.ConnectionParameters(
            host = self.configuration['Host'],
            virtual_host = self.configuration['VirtualHost'],
            credentials = credentials
        ))

        self.channel = self.connection.channel()

        logger.info("MESSAGEQUEUE - Initialized.")

    def StartConsuming(self, StoreMessage):
        self.channel.basic_consume(
            self.configuration['Queue'],
            StoreMessage
        )

        self.channel.start_consuming()

        logger.info("MESSAGEQUEUE - Started consuming.")

    def Finalize(self):
        self.channel.stop_consuming()
        self.connection.close()

        logger.info("MESSAGEQUEUE - Finalized consuming.")